appservice/src/SiteWrapper.ts

```diff
@@ -293,6 +298,43 @@ export class SiteWrapper {
         await requestP.get(`https://${this.defaultHostName}/admin/host/status`, requestOptions);
     }
 
+    public async validateDeployment(outputChannel: vscode.OutputChannel, pollingInterval: number = 5000, pollingAttempts: number = 12): Promise<responseStatus[]> {
+        // by default, try for a minute before we declare that the deployment failed
```

Comment:
Now that I think about it more - I would prefer if it was based on time, not number of attempts. Using "number of attempts" would have inconsistent results dependent on how long it takes to poll the site each time. For example, if it takes 10 seconds to poll the site, the user is actually waiting (5 + 10)*12=3 minutes

I'd prefer something like this:
```typescript
const maxTime: number = Date.now() + timeoutInSeconds * 1000;
while (Date.now() < maxTime) {
    ...
}
```